internal/bminventory/inventory.go

```diff
@@ -354,6 +354,22 @@ func (b *bareMetalInventory) validateRegisterClusterInternalParams(params *insta
 	return nil
 }
 
+func (b *bareMetalInventory) validateAccessToARMFunctionality(ctx context.Context, cpuArchitecture string) error {
+	var err error
+	var armArchAllowed bool
+
+	if cpuArchitecture == "arm64" {
+		armArchAllowed, err = b.authzHandler.HasOrgBasedCapability(ctx, ocm.ArmCapabilityName)
+		if err != nil {
+			return fmt.Errorf("error getting user %s capability, error: %w", ocm.ArmCapabilityName, err)
+		}
+		if !armArchAllowed {
+			return common.NewApiError(http.StatusBadRequest, err) // TODO: Check which HTTP return code is expected here.
+		}
```

Comment:
lines 362-368 can be extracted to some generic helper.
e.g.
checkOrgBasedCapability(ctx context.Context, capabilityName string) error

Reply:
this is what I tried to do with `authzHandler.HasOrgBasedCapability`, this is just a nice wrapper that handles err and return code

Reply:
yeah, just meant that if we can refactor this func to switch/case, a generic helper can be used to handle error flows.

Reply:
please check if looks okay to you now.